server/auth.cpp

```diff
@@ -35,11 +35,6 @@ _   ._       Copyright (c) 1996-2021 Freeciv21 and Freeciv contributors.
 #define GUEST_NAME "guest"
 
 #define MIN_PASSWORD_LEN 6 // minimum length of password
```

Comment:
Cant we remove MIN_PASSWORD_LEN and is_good_password() entirely (just check for ascii and garbage and keep the function as is_password_valid() or something) ? If we dont want to have custom servers for password quality, we shouldnt bother with length too imo.

Reply:
Let's say that such a discussion is out of the scope of this PR which is to fix Coverity warnings about always-true conditionals. We can open an issue if we want to remove it.